src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -223,6 +229,145 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
         }
     }
 
+
+    @DataProvider()
+    Object[][] mergeIsDafeProvider() {
+        return new Object[][]{
+                new Object[]{new HaplotypeProbabilitiesFromGenotype(snp, hb, 5D, 0D, 10D), new HaplotypeProbabilitiesFromGenotype(snp, hb, 0D, 10D, 100D)},
+                new Object[]{
+                        new HaplotypeProbabilityOfNormalGivenTumor(
+                                new HaplotypeProbabilitiesFromGenotype(snp, hb, 5D, 0D, 10D), .05),
+                        new HaplotypeProbabilityOfNormalGivenTumor(
+                                new HaplotypeProbabilitiesFromGenotype(snp, hb, 0D, 10D, 100D), 0.05)},
+                new Object[]{new HaplotypeProbabilitiesFromSequence(hb, 5, 0, 1), new HaplotypeProbabilitiesFromSequence(hb, 3, 2, 1)},
+                new Object[]{new HaplotypeProbabilitiesFromContaminatorSequence(hb, .1, 5, 0, 1), new HaplotypeProbabilitiesFromContaminatorSequence(hb, .1, 5, 1, 0)},
+        };
+    }
+
+    @Test(dataProvider = "mergeIsDafeProvider")
+    public void testMergeHaplotypeProbabilitiesIsSafe(final HaplotypeProbabilities hp1, final HaplotypeProbabilities hp2){
+
+        final HaplotypeProbabilities merged1 = hp1.deepCopy().merge(hp2);
+        final HaplotypeProbabilities merged2 = hp1.deepCopy().merge(hp2);
+
+        Assert.assertEquals(merged1.getLikelihoods(),merged2.getLikelihoods());
+    }
+
+
+    @Test(dataProvider = "mergeIsDafeProvider")
+    public void testMergeFingerprintIsSafe(final HaplotypeProbabilities hp1, final HaplotypeProbabilities hp2){
+
+        final Fingerprint fpA = new Fingerprint("test2",null,"none");
+        final Fingerprint fpB = new Fingerprint("test2",null,"none");
+
+        final Fingerprint fp1 = new Fingerprint("test1",null,"none");
+        fp1.add(hp1);
+
+        final Fingerprint fp2 = new Fingerprint("test1",null,"none");
+        fp2.add(hp2);
+
+        fpA.merge(fp1);
+        fpB.merge(fp1);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertEquals(fpA.get(hb).getLikelihoods(),fpB.get(hb).getLikelihoods());
+        }
+
+        fpA.merge(fp2);
+        fpB.merge(fp2);
+        fpB.merge(fp2);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertNotEquals(fpA.get(hb), fpB.get(hb));
+        }
+
+        fpA.merge(fp2);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertEquals(fpA.get(hb).getLikelihoods(),fpB.get(hb).getLikelihoods());
+        }
+    }
+
+
+    @Test
+    public void testMergeIsSafeFromSequence() {
+        final Path na12891_r1 = new File(TEST_DATA_DIR, "NA12891.over.fingerprints.r1.sam").toPath();
```

Comment:
I think a more idiomatic way to write this is
```suggestion
        final Path na12891_r1 = Paths.get(TEST_DATA_DIR).resolve("NA12891.over.fingerprints.r1.sam");
```

Reply:
since TEST_DATA_DIR is a File, it's actually more lengthy as I need to stick a `getAbsoluteFileName()` in there as well....I'll do it if you prefer though.

Reply:
Under what circumstances is `getAbsoluteFileName()` required? I would think that `Path` would work the same with or without an absolute filename.

Reply:
Oh, I thought it was `String`. If it's `File` then you can write:
```
        final Path na12891_r1 = TEST_DATA_DIR.toPath().resolve("NA12891.over.fingerprints.r1.sam");
```
I don't know why you'd need to use `getAbsoluteFileName()` in this code.